tunnel/options.go

```diff
@@ -29,6 +31,15 @@ type Options struct {
 	Transport transport.Transport
 }
 
+type DialOption func(*DialOptions)
+
+type DialOptions struct {
```

Comment:
If you decide to turn the `cast`* options into some enum, this would have to be changed 🤔

Reply:
Related to the other point. There should probably be a cleaner way but depends on whether we support multiple messaging types.